source/slides/05_linux_basics.rst

```diff
@@ -0,0 +1,299 @@
+.. _05_linux_basics:
+
+Storage & Automated OS Install
+==============================
+
+Storage
+=======
+
+Virtualizing storage
+--------------------
+
+* Partitions have limitations
+* Storage needs to be flexible
+* Large storage management
+
+  * Adding, removing, expanding storage hot swap
+
+Utilizing storage more efficiently
+----------------------------------
+
+* Grow the storage as needed
+* Only allocate the minimum you need
+* Expect to make changes to your storage later
+* Fully utilize the physical media
+
+Storage Snapshots
+-----------------
+
+* Perform consistent backups atomically
+
+LVM
+---
+
+Logical Volume Manager
+
+.. figure:: ../_static/lvm.svg
+  :align: center
+  :width: 60%
+
+Components of LVM
+-----------------
+
+.. csv-table::
+  :header: Name, Description
+
+  Physical Volume (PV), physical device (typically a hdd)
+  Volume Group (VG), Collection of PV's
+  Logical Volume (LV), Slice of a VG to create a partition
+  Physical Extent (PE), Chunks of data in a PV with the same size
+  Logical Extent (LE), Chunks of data in a LV with the same size
+
+`LVM Howto`_ (a bit dated, but still relevant)
+
+.. _LVM Howto: http://tldp.org/HOWTO/LVM-HOWTO/
+
+LVM Example
+-----------
+
+.. rst-class:: build
+
+#. Two disk partitions: sda1 (800MB) & sdb1 (400MB)
+#. Format each partition into a PV
+#. The PV's are divided up into 4MB chunks (PE): sda1 (200PE) / sdb1 (100PE)
+#. Create a VG using sda1 and sda2
+#. Create an LV of size 1GB (uses linear mapping by default)
+#. Total of 250 LE's are required
+#. 200PE's are used in sda1, 50PE's are used in sdb1, 50PE are free
+#. Create a filesystem from the LV and mount it
+#. PROFIT!
+
+LVM Visualized
+--------------
+
+.. insert graphic here
+
+LVM commands
+------------
+
+.. code-block:: bash
+
+  $ yum install lvm2
+
+  $ pvcreate /dev/loop1 /dev/loop2
```

Comment:
Yes, I didn't want to give the exact commands since that's part of the hw assignment :) But I can add a comment basically mentioning that.